src/hubble_ble.c

```
@@ -140,7 +140,12 @@ int hubble_ble_init(uint64_t utc_time)
 
 	HUBBLE_LOG_INFO("Hubble BLE Network initialized\n");
 
-	utc_time_base = utc_time - hubble_uptime_get();
+	if (utc_time != 0) {
+		hubble_ble_utc_set(utc_time);
+	}
+	if (key) {
+		hubble_ble_key_set(key);
+	}
```

I think we should return an error if these parameters are not valid .. 

The reason is that the stack can't be used if there is no key or time, we can assume that `key_set` and `utc_set` are to update / change these parameters.

---

Do we want to update that one app then to not init until we get the key? The only reason I did this is from that one sample that hubble_ble_init then sets the key later. I agree that we should just have it on init and am willing to change this.

---

I think so ... the contract is that the stack is usable after init() and that is not the case until a key/utc

---

it should return `-EINVAL` case one of these parameters are missing